eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -593,14 +595,27 @@ def param(self, matches):
         return Tree("param", matches)
 
     def model(self, matches):
+        rules = set()
         definitions = dict()
         regulation = None
         inits = collections.Counter()
         for match in matches:
             if type(match) == dict:
                 key, value = list(match.items())[0]
                 if key == 'rules':
-                    rules = set(value)
+                    rules.update(value)
+                if key == 'inits':
+                    inits = value
+                if key == 'definitions':
+                    definitions = value
+                if key == 'regulation':
+                    regulation = value
```

Comment:
Maybe there should be `dict.update` again somewhere?

Reply:
Yes, `dict.update` solved the issue in commit 8546c4e. 
Commit 1ba38d2 handles multiple regulation definitions by throwing an exception.